interview-questions.md

```diff
@@ -8,36 +8,67 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. As a developer, I am creating a Rails application with a model called Cohort that has_many students, but OOPS! I forgot to add the foreign key. How can I fix this mistake? What is the name of the foreign key? Would the foreign key be on the Cohort model or the Student model?
 
-Your answer:
+Your answer: Sounds like it's time for a new migration! Blanking off the pseudo of the top of my head, but you'd do something like [rails generate migration add_foreign_key], and then go into the migration file and put whatever code is necessarry to add your foreign key there. as for the name and where it is, I THINK it should be [student_id], in the cohort model.
 
-Researched answer:
+Researched answer: commands should be [rails generate migration add_foreign_key_to_students cohort:references] to create the migration file, [def change
+  add_foreign_key :students, :cohorts
+end
+], inside the migration file to do the adding, and then raild db:migrate back in the console to update the database! Oh, and I got the other two questions backwards. it should be [cohort_id], in the student model. ***.tnemom gniniarb sdrawkcab a dah***
+**this looks so strange with a code example in it. hard to read when everything is the same color >_<**
 
 2. Which RESTful routes must always be passed params? Why?
 
-Your answer:
+Your answer: Delete springs to mind, since you can't really delete nothing unless it can just wipe an entire database. Put 'n Patch also specifically target existing data so would be pretty useless without params methinks.
 
-Researched answer:
+Researched answer: **researching this one hurts. feels like I could spend days looking into this and still barely understand much more than "technically none of them but also all of 'em depending on the use case".**
 
 3. Name three rails generator commands. What is created by each?
 
-Your answer:
+Your answer:Rails g model, resource, and controller are the three that I can recall. as for what they do specifically, model creates the basic layout for entering data into a database, resource Does the same thing but also adds HTTP verbage, and controller makes the bones of a controller file.
 
-Researched answer:
+Researched answer: Can't think of much else to add, sorry.
```

Comment:
There is scaffold and a few others out there. One of the things to keep in mind is what files they create and are they necessary.